### PR TITLE
CME-192: migrate springfox to spring doc

### PR DESCRIPTION
**Pull request by AntonyLeons**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CME-192

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/hmc/controllers/HearingActualsController.java [hunk 1]
```diff
@@ -27,22 +27,23 @@ public class HearingActualsController {
...
     @ResponseStatus(HttpStatus.NO_CONTENT)
-    @ApiResponses(value = {
-        @ApiResponse(code = 204, message = "Hearing id is valid"),
-        @ApiResponse(code = 404, message = ValidationError.HEARING_ID_NOT_FOUND),
-        @ApiResponse(code = 400, message = ValidationError.INVALID_HEARING_ID_DETAILS)
+    @ApiResponses({
+        @ApiResponse(responseCode = "204", description = "Hearing id is valid"),
+        @ApiResponse(responseCode = "404", description = ValidationError.HEARING_ID_NOT_FOUND),
+        @ApiResponse(responseCode = "400", description = ValidationError.INVALID_HEARING_ID_DETAILS)
     })
```
Comment (rapolaskaseliscgi) [inline src/main/java/uk/gov/hmcts/reform/hmc/controllers/HearingActualsController.java:41]: I'd suggest to remove wrapper, and leave just this:
```
@ApiResponse(responseCode = "204", description = "Hearing id is valid")
@ApiResponse(responseCode = "404", description = ValidationError.HEARING_ID_NOT_FOUND)
@ApiResponse(responseCode = "400", description = ValidationError.INVALID_HEARING_ID_DETAILS)
```
I believe it is picked up by sonar as a code smell.

File: src/main/resources/application.yaml [hunk 2]
```diff
@@ -57,7 +57,9 @@ spring:
       client:
         composite-indicator:
           enabled: false
-
+  mvc:
+    pathmatch:
+      matching-strategy: ant_path_matcher
```
Comment (rapolaskaseliscgi) [inline src/main/resources/application.yaml:62]: Is this needed for Spring Boot 2 update? I am probably wrong, but in my understanding this change was introduced only in Spring Boot 3.

Comment (AntonyLeons) [reply src/main/resources/application.yaml]: think you're right, probably i read ahead in the migration guide

File: src/main/java/uk/gov/hmcts/reform/hmc/controllers/HearingActualsController.java (hunk 1 above)
Comment (AntonyLeons) [reply src/main/java/uk/gov/hmcts/reform/hmc/controllers/HearingActualsController.java:41]: its best practice, and passes test so i think ill leave it in, also keeps diff smaller

Comment (rapolaskaseliscgi) [reply src/main/java/uk/gov/hmcts/reform/hmc/controllers/HearingActualsController.java:41]: I don't see how is it a best practice if it is marked as code smell.. This was required with springfox but is undesirable (though allowed) with springdoc (https://github.com/joansmith/sonar-java/blob/master/java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S1710.html). I am fine either way.

**Review by RebeccaBaker — APPROVED** [review]

**Pull request merged**